autowisp/browser_interface/configuration/views.py

```diff
@@ -31,10 +31,65 @@
 
 # pylint: enable=no-name-in-module
 
+def _merge_children(old_children, new_children, parent_type=None):
+    """Merge lists of child nodes by (name,type); keep order, append new."""
+    old_children = old_children or []
+    new_children = new_children or []
+
+    # For parameter nodes the child is usually a single 'value' node.
+    if parent_type == "parameter":
+        # If the new file supplied a value, use it; otherwise keep old.
+        return new_children if new_children else old_children
+
+    # Generic case: merge by (name, type)
+    merged = list(old_children)
+    index = {
+        (c.get("name"), c.get("type")): i
+        for i, c in enumerate(old_children)
+        if isinstance(c, dict)
+    }
+
+    for child in new_children:
+        if isinstance(child, dict):
+            key = (child.get("name"), child.get("type"))
+            if key in index:
```

Comment:
It is a problem if a parameter is not in `index` that means that the current pipeline does not expect a parameter with that name to exist. so this should just be `assert key in index`.